### ISSUE TITLE
Hope to get the project file to generate DLL,

### ISSUE THREAD

**Issue by zhaohuitao-123** [PetteriAimonen/focus-stack#52]
sorry,I Hope to get the project file to generate DLL,But now only as a console program。

**Comment by PetteriAimonen** [comment]
You may want to wrap it in C API for best ABI compatibility.
But other than that, compiling focus-stack to DLL is no different than compiling any other C++ program to DLL.

**Comment by zhaohuitao-123** [comment]
thank you for your reply， I'm a newcomer.  In windwos 10 ，I had nmake the project。  but it was .exe after nmake

**Comment by PetteriAimonen** [comment]
Yes, the premade build rules do not make a DLL. You'll have to write your own.

**Comment by zhaohuitao-123** [comment]
thanks。Dll can be used。  A new challenge, focusing on 2 images (500M pixels, taken by a microscope) took almost 1 second, it is too long. I want to optimize  to less than 10ms. Do you have a better suggestion

**Comment by PetteriAimonen** [comment]
You probably won't get it that fast, at least not with this algorithm.

Verify that you have OpenCL support working for your graphics card, that helps a little. But those images are large and will take time to process.

Focus-stack parallelism is currently only image level, so with only 2 input images you are using at most 2 CPU cores. Using the C++ API you can add new images while previous ones are being processed. Make sure to add the images through memory buffers (`FocusStack::add_image(const cv::Mat &image)`) to avoid spending time loading them from drive.

**Comment by zhaohuitao-123** [comment]
I had loading them from memory。
if  the image provided had corrected and aligned，
only merge those by Focus-stack, so how long it will takes you think?
and after this, feature level fusion and pixel level fusion which one you think faster？

**Comment by PetteriAimonen** [comment]
You can see individual step timings in the verbose log.

But overall, your 10 ms target with 500 Mpixel images sounds like something that can only be achieved by highly optimized and specialized algorithm running on a powerful GPU.

**Comment by zhaohuitao-123** [comment]
I apologize for bothering you again. Due to some reasons, I need to modify the waveform of wavelet transform. In the file "task_wavelet_templates.hh", "c_lospass" & "c_hipass", I searched articles about Daubechies Wavelet, but couldn't find the source of your data. Could you please tell me?

**Comment by zhaohuitao-123** [comment]
I apologize for bothering you again. Due to some reasons, I need to modify the waveform of wavelet transform. In the file "task_wavelet_templates.hh", "c_lospass" & "c_hipass", I searched articles about Daubechies Wavelet, but couldn't find the source of your data. Could you please tell me?

**Comment by PetteriAimonen** [comment]
"Image Processing with Complex Daubechies Wavelets", J.M. Lina, 1997
or equivalently:
"Complex Daubechies Wavelets: Filters Design and Applications", J.-M. Lina, 1997

The latter is available from https://www.researchgate.net/publication/2745198_Complex_Daubechies_Wavelets_Filters_Design_and_Applications
